website/thaliawebsite/views.py

```diff
@@ -48,9 +49,19 @@ def get_context_data(self, **kwargs) -> dict:
 
         page_range = range(page_range_start, page_range_stop)
 
+        # generates the string of keywords, and applies these to the base_url
+        keyword_string = ""
+
+        for keyword in self.keywords:
+            if keyword_string == "":
+                keyword_string += f"{keyword}"
+            else:
+                keyword_string += f"+{keyword}"
+
         context.update(
             {
                 "page_range": page_range,
+                "base_url": self.request.path + f"?keywords={keyword_string}",
```

Comment:
Sorry I didnt explain properly. This needs to work with _any_ query parameters, not specifically the ones used in albums (because there are other PagedViews with other parameters. 
It should probably be something like this:
- Make a copy of request.GET, using QueryDict.copy() (see https://docs.djangoproject.com/en/5.0/ref/request-response/ for how it all works).
- Then in that copy, remove the 'page' key if it exists. 
- Then make the base_url by using request.path + QueryDict.urlencode().

(this urlencode part is important for security. As it is right now, query parameters are not escaped, so you can abuse it to e.g. include JS)